src/tcgwars/logic/impl/gen8/BlueSkyStream.groovy

```diff
@@ -195,13 +196,29 @@ public enum BlueSkyStream implements LogicCardInfo {
         weakness R
         bwAbility "Solar Evolution", {
           text "When you attach an Energy from your hand to this Pokémon during your turn, you may use this Ability. Search your deck for a card that evolves from this Pokémon and put it onto this Pokémon to evolve it. Then, shuffle your deck."
-          actionA {
+          delayedA {
+            after ATTACH_ENERGY, self, {
+              if (ef.reason == PLAY_FROM_HAND
+                && bg.currentTurn == self.owner
+                && self.owner.pbg.deck
+                && lastUsedTurn != bg.turnCount
```

Comment:
This ability can be used multiple times in a turn.

Reply:
Please check that the new logic makes sense.